chain/epoch-manager/src/shard_assignment.rs

```diff
@@ -135,11 +135,6 @@ fn assign_with_possible_repeats<T: HasStake + Eq, I: Iterator<Item = (usize, T)>
     }
 }
 
-/// Marker struct to communicate the error where you try to assign validators to shards
-/// and there are not enough to even meet the minimum per shard.
-#[derive(Debug)]
-pub struct NotEnoughValidators;
```

Comment:
why remove this instead of adding another error type (if this is the reason)?

Reply:
Because there is only one reason for error, so we can return `Err(())`, I don't see why marker is needed. We can also return EpochError, I don't have strong opinion on it.

Reply:
IMO errors should be as unique as possible. I would suggest either keeping the tag or adding a string message to the new error. You may also want to improve it a bit and add the relevant numbers to it (`num_chunk_producers`, `min_validators_per_shard`).

Reply:
Just reverted change for now.